[WebGPU] Move C++ to Swift interop declarations out of public module

emw-apple opened the pull request:
#### 38af50008577182013ebd31319cdaffd13ba5ee6
<pre>
[WebGPU] Move C++ to Swift interop declarations out of public module
<a href="https://bugs.webkit.org/show_bug.cgi?id=283152">https://bugs.webkit.org/show_bug.cgi?id=283152</a>
<a href="https://rdar.apple.com/problem/139929679">rdar://problem/139929679</a>

Reviewed by Mike Wyrzykowski.

The Swift module built in WebGPU needs some special declarations in
order to compile (for example, specializations of std::span). Currently
these are declared in the public &quot;WebGPU&quot; module, which means that they
are public headers and are accessible to clients.

Clean this up by adding a &quot;WebGPU_Internal&quot; module following the recent
precedent in WebKit (<a href="https://commits.webkit.org/286465@main).">https://commits.webkit.org/286465@main).</a> This
module serves two purposes inside WebGPU: Swift code imports it to
access non-API declarations, and C++ code imports it to access Swift&apos;s
generated declarations. For now, the module exports
&quot;WebGPUSwiftInternal.h&quot;, but other project-level headers can be added as
needed.

Move the std::span specializations to WebGPUSwiftInternal.h. Future use
of WebGPUSwift.h will be used for Swift-related declarations that
clients of WebGPU need to know about.

* Source/WebGPU/Configurations/WebGPU.xcconfig:
* Source/WebGPU/WebGPU.xcodeproj/project.pbxproj:
* Source/WebGPU/WebGPU/Buffer.swift:
* Source/WebGPU/WebGPU/Internal/WebGPUSwiftInternal.h: Renamed from Source/WebGPU/WebGPU/WebGPUSwiftInternal.h.
* Source/WebGPU/WebGPU/Queue.swift:
* Source/WebGPU/WebGPU/WebGPU.modulemap:
* Source/WebGPU/WebGPU/WebGPUSwift.h:

Canonical link: <a href="https://commits.webkit.org/286661@main">https://commits.webkit.org/286661@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/8806803d3f4c6cb2bccca11f333e193d33c1f9a7

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/76592 "1 style error") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/55627 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/29498 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/81122 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/27868 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/64769 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/3920 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/60045 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/18146 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/79659 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/49963 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/65762 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/40370 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/47363 "Passed tests") | | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/26192 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/68485 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/23586 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/82568 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/3968 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/2612 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/68323 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/4121 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/65734 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/67572 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/16870 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/11541 "Passed tests") | [❌ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/9627 "Found 1 new test failure: http/tests/local/blob/navigate-blob.html (failure)") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/3915 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/3938 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/7368 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/5696 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

mwyrzykowski reviewed (approved):
Looks good from my side, thank you!

emw-apple committed:
[WebGPU] Move C++ to Swift interop declarations out of public module
https://bugs.webkit.org/show_bug.cgi?id=283152
rdar://problem/139929679

Reviewed by Mike Wyrzykowski.

The Swift module built in WebGPU needs some special declarations in
order to compile (for example, specializations of std::span). Currently
these are declared in the public "WebGPU" module, which means that they
are public headers and are accessible to clients.

Clean this up by adding a "WebGPU_Internal" module following the recent
precedent in WebKit (https://commits.webkit.org/286465@main). This
module serves two purposes inside WebGPU: Swift code imports it to
access non-API declarations, and C++ code imports it to access Swift's
generated declarations. For now, the module exports
"WebGPUSwiftInternal.h", but other project-level headers can be added as
needed.

Move the std::span specializations to WebGPUSwiftInternal.h. Future use
of WebGPUSwift.h will be used for Swift-related declarations that
clients of WebGPU need to know about.

* Source/WebGPU/Configurations/WebGPU.xcconfig:
* Source/WebGPU/WebGPU.xcodeproj/project.pbxproj:
* Source/WebGPU/WebGPU/Buffer.swift:
* Source/WebGPU/WebGPU/Internal/WebGPUSwiftInternal.h: Renamed from Source/WebGPU/WebGPU/WebGPUSwiftInternal.h.
* Source/WebGPU/WebGPU/Queue.swift:
* Source/WebGPU/WebGPU/WebGPU.modulemap:
* Source/WebGPU/WebGPU/WebGPUSwift.h:

Canonical link: https://commits.webkit.org/286661@main
